src/Middleware/Diagnostics/test/UnitTests/ExceptionHandlerMiddlewareTest.cs

```diff
@@ -315,6 +315,47 @@ public async Task Metrics_ExceptionThrown_Unhandled_Reported()
             m => AssertRequestException(m, "System.InvalidOperationException", "unhandled"));
     }
 
+    [Fact]
+    public async Task ExceptionFeatureSetOnDeveloperExceptionPage()
+    {
+        // Arrange
+        using var host = new HostBuilder()
+            .ConfigureWebHost(webHostBuilder =>
+            {
+                webHostBuilder
+                .UseTestServer()
+                .Configure(app =>
+                {
+                    app.Use(async (context, next) =>
+                    {
+                        await next();
+
+                        var exceptionHandlerFeature = context.Features.GetRequiredFeature<IExceptionHandlerPathFeature>();
+
+                        Assert.NotNull(exceptionHandlerFeature);
+                        Assert.Equal("Test exception", exceptionHandlerFeature.Error.Message);
+                        Assert.Equal(context.Request.Path, exceptionHandlerFeature.Path);
+                    });
+                    app.UseExceptionHandler(exceptionApp =>
+                    {
+                        exceptionApp.Run(context => Task.CompletedTask);
+                    });
+                    app.Run(context =>
+                    {
+                        throw new Exception("Test exception");
+                    });
+
+                });
+            }).Build();
+
+        await host.StartAsync();
+
+        var server = host.GetTestServer();
+        var request = new HttpRequestMessage(HttpMethod.Get, "/path");
+
+        var response = await server.CreateClient().SendAsync(request);
```

Comment:
This test isn't forcing the asserts to run. Move asserts into the main path of the test:

```suggestion
        // Arrange
        var tcs = new TaskCompletionSource<IExceptionHandlerPathFeature>(TaskCreationOptions.RunContinuationsAsynchronously);

        using var host = new HostBuilder()
            .ConfigureWebHost(webHostBuilder =>
            {
                webHostBuilder
                .UseTestServer()
                .Configure(app =>
                {
                    app.Use(async (context, next) =>
                    {
                        await next();
                        
                        var exceptionHandlerFeature = context.Features.GetRequiredFeature<IExceptionHandlerPathFeature>();
                        tcs.SetResult(exceptionHandlerFeature);
                    });
                    app.UseExceptionHandler(exceptionApp =>
                    {
                        exceptionApp.Run(context => Task.CompletedTask);
                    });
                    app.Run(context =>
                    {
                        throw new Exception("Test exception");
                    });

                });
            }).Build();

        await host.StartAsync();

        var server = host.GetTestServer();
        var request = new HttpRequestMessage(HttpMethod.Get, "/path");

        var response = await server.CreateClient().SendAsync(request);

        var feature = await tcs.Task;
        Assert.NotNull(feature);
        Assert.Equal("Test exception", feature.Error.Message);
        Assert.Equal(context.Request.Path, feature.Path);
```

(note: changed in a text field, not might not compile, and/or have bugs)